packages/meditrak-server/src/dataAccessors/updateOrCreateSurveyResponse.js

```diff
@@ -66,19 +66,7 @@ export async function updateOrCreateSurveyResponse(models, surveyResponseObject)
     }
 
     // Ensure data_time is populated, supporting legacy versions of MediTrak
-    const {
-      data_time: suppliedDataTime,
-      submission_time: submissionTime, // v1.7.87 to v1.9.110 (inclusive) uses submission_time
-      end_time: endTime, // prior to v1.7.87 fall back to end_time
-      timezone = DEFAULT_DATABASE_TIMEZONE, // if no timezone provided, use db default
-    } = surveyResponseObject;
-    const dataTime = suppliedDataTime || submissionTime || endTime;
-
-    // Convert to the original timezone, then strip timezone suffix, so it ends up in db as it
-    // appeared to the original survey submitter
-    surveyResponseProperties.data_time = stripTimezoneFromDate(
-      momentTimezone(dataTime).tz(timezone).format(),
-    );
+    surveyResponseProperties.data_time = getDataTime(surveyResponseObject);
```

Comment:
With this change, I wanted to tackle the problematic business logic of silently defaulting to Pacific/Auckland. 

It took me a couple days of sleuthing to figure out what was going on, and I am very familiar with the codebase, it might take the next person a week or more. Having it default to Pacific/Auckland is not obvious. Also, it introduces inconsistency because I would not have run into the problem if I were developing in NZ, but I just so happened to be in Melb, in a different timezone.

The first option would be to throw an error if timezone is missing, but this doesn't make sense as an interface. Our clients e.g. psss-server already have the data_time in the correct format (iso8601 without tz), they shouldn't be inventing a timezone, which will then be stripped.

So our second option, and the one I settled on, was to change the behaviour to only default to Pacific/Auckland for older clients as a fallback. Modern clients which use data_time will never have the timezone defaulted, they can either specify it, or specify a valid known data_time string in format ISO_8601 without tz.

Reply:
@EMcQ-BES Would be great to get your input on this, as you worked on this recently

Reply:
This all definitely **looks** like a very reasonable update to the `data_time` logic, removing an assumption of the original design and making it more generic is always a good call.

I'm definitely interested in seeing some regression testing since it's always a little difficult to wrap your head around timezone logic in code.

Reply:
I like this change too! I think at the moment we only have meditrak-app and psss-server clients for this endpoint. So hopefully regression testing is not going to be too much...

Reply:
As discussed we'll also bring the same change to `importSurveyResponses` endpoint